x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/InsensitiveEquals.java

```diff
@@ -83,6 +85,10 @@ protected TypeResolution resolveType() {
     }
 
     public static Automaton automaton(BytesRef val) {
+        if (val.length == 0) {
+            // toCaseInsensitiveString doesn't match empty strings properly so let's do it ourselves
+            return Automata.makeEmptyString();
```

Comment:
_search works around this as well with:
```
        // check if valueForSearch is the same as an empty string
        // if we have a length of zero, just do a regular term query
        if (valueForSearch.length == 0) {
            return termQuery(value, context);
        }
```
Which makes sense. And it's slightly faster. Here, instead of doing a term query I just make an automata for the empty string. We are *already* pushing a proper term query using the _search infrastructure when possible.